libraries/AP_Motors/AP_MotorsSingle.cpp

```diff
@@ -23,22 +23,27 @@ extern const AP_HAL::HAL& hal;
 // init
 void AP_MotorsSingle::init(motor_frame_class frame_class, motor_frame_type frame_type)
 {
-    // make sure 6 output channels are mapped
-    for (uint8_t i = 0; i < 6; i++) {
-        add_motor_num(CH_1 + i);
-    }
+    set_initialised_ok(false);
+    if (frame_class == MOTOR_FRAME_SINGLE) {
```

Comment:
If this isn't the case, why were we called?

I was expecting a simple `const char * get_frame_class() { return "Single"; }` in the header...

Reply:
I didn't think to question why it was written this way, but I agree that for this subclass, it's better to assign the string in the header, and revert the cpp to its original state.